lib/storage/bucket.js

```diff
@@ -365,25 +370,38 @@ Bucket.prototype.file = function(name) {
  * }, function(err, files, nextQuery) {});
  */
 Bucket.prototype.getFiles = function(query, callback) {
-  var that = this;
+  var self = this;
+
   if (!callback) {
     callback = query;
     query = {};
   }
+
   this.makeReq_('GET', '/o', query, true, function(err, resp) {
     if (err) {
       callback(err);
       return;
     }
+
     var files = (resp.items || []).map(function(item) {
-      var file = that.file(item.name);
+      var options = {};
+
+      if (query.versions) {
+        options.generation = item.generation;
+      }
```

Comment:
If a user specified that they want to retrieve all the versions of a file, they will get back files scoped to their generation.